### ISSUE TITLE
npm test intermittently fails with - ParserError: Expected ';' but got 'ILLEGAL'

### ISSUE THREAD

**Issue by peterbroadhurst** [trufflesuite/truffle#2225]
- [ ] I've asked for help in the [Truffle Gitter](http://gitter.im/Consensys/truffle) before filing this issue.

---------------------------

## Issue

Hit the following error from `truffle test` from automation:
```
Error: CompileError: truffle/DeployedAddresses.sol:4:75: ParserError: Expected ';' but got 'ILLEGAL'
function Migrations() public pure returns (address payable) { return 0.5.80d0c2BF6239b1E58211E0B6510e1Fa1650717; }
```

The problem is that this regex can match `0<anything>5<anything>0` in the ethereum addresss, and replace that with the current compiler version such as `0.5.8`:
https://github.com/trufflesuite/truffle/blob/9acc723e6942fa059290fc7cbcfc3e83110bd444/packages/truffle-core/lib/testing/deployed.js#L42

My suggestion for a fix is to change the regex to `source.replace(/0\.5\.0/gm, v);` - I will try and submit a PR along side this issue.

Here's some quick node output to show what I mean:
```
> let a = '0a5b0c'; // this clearly isn't a solc version
undefined
> a.replace(/0.5.0/gm, '0.5.8'); // but this still replaces the `a` and `b`
'0.5.8c'
> a.replace(/0\.5\.0/gm, '0.5.8'); // where as this works fine
'0a5b0c'
> let b = '0.5.0'
undefined
> b.replace(/0\.5\.0/gm, '0.5.8'); // and still replaces an actual solc version
'0.5.8'
```

## Steps to Reproduce

It is a random occurrence, so you would need to run truffle test a lot of times to hit this.

## Expected Behavior

`trufle test` runs

## Actual Results

```
Error: CompileError: truffle/DeployedAddresses.sol:4:75: ParserError: Expected ';' but got 'ILLEGAL'
function Migrations() public pure returns (address payable) { return 0.5.80d0c2BF6239b1E58211E0B6510e1Fa1650717; }
```

## Environment

* Operating System: Linux
* Ethereum client: Geth/PoA (in Kaleido)
* Truffle version (`truffle version`): 0.5.27
* node version (`node --version`): v10.15.3
* npm version (`npm --version`): 6.7.0

**Comment by CruzMolina** [comment]
Thanks for opening this issue @peterbroadhurst ! Are you still interested in opening a PR for this?

**Comment by peterbroadhurst** [comment]
Sorry @CruzMolina I've not go this raised yet.
We've seen this again today in our pipeline, which reminded me. I will go ahead and get the PR raised now.

**Comment by haltman-at** [comment]
Oops, seems we never closed this.  (This was released in v5.0.30.)  Closing!
